scr/main/java/bittercode/util/DatabaseConfig.java

```diff
@@ -0,0 +1,30 @@
+package com.bittercode.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+class DatabaseConfig {
+
+    static Properties prop = new Properties();
+    static {
+
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        InputStream input = classLoader.getResourceAsStream("application.properties");
+
+        try {
+            prop.load(input);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public final static String DRIVER_NAME = prop.getProperty("db.driver");
+    public final static String DB_HOST = prop.getProperty("db.host");
+    public final static String DB_PORT = prop.getProperty("db.port");
+    public final static String DB_NAME = prop.getProperty("db.name");
+    public final static String DB_USER_NAME = prop.getProperty("db.username");
+    public final static String DB_PASSWORD = prop.getProperty("db.password");
+    public final static String CONNECTION_STRING = DB_HOST + ":" + DB_PORT + "/" + DB_NAME;

```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/2#issuecomment-2418799949>#b11fc3</a><span> - 10/17/2024, 08:03 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The database configuration constants are currently declared as public. This approach might expose sensitive information and make it difficult to change the implementation in the future. Have you considered encapsulating these fields and providing getter methods if external access is required?
</div>
<div id="fix">
<b>Fix:</b> Change the access modifier of the configuration constants to private and provide public getter methods if external access is required.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -21,10 +21,10 @@
      }
 
 -    public final static String DRIVER_NAME = prop.getProperty("db.driver");
 -    public final static String DB_HOST = prop.getProperty("db.host");
 -    public final static String DB_PORT = prop.getProperty("db.port");
 -    public final static String DB_NAME = prop.getProperty("db.name");
 -    public final static String DB_USER_NAME = prop.getProperty("db.username");
 -    public final static String DB_PASSWORD = prop.getProperty("db.password");
 -    public final static String CONNECTION_STRING = DB_HOST + ":" + DB_PORT + "/" + DB_NAME;
 +    private final static String DRIVER_NAME = prop.getProperty("db.driver");
 +    private final static String DB_HOST = prop.getProperty("db.host");
 +    private final static String DB_PORT = prop.getProperty("db.port");
 +    private final static String DB_NAME = prop.getProperty("db.name");
 +    private final static String DB_USER_NAME = prop.getProperty("db.username");
 +    private final static String DB_PASSWORD = prop.getProperty("db.password");
 +    private final static String CONNECTION_STRING = DB_HOST + ":" + DB_PORT + "/" + DB_NAME;
 
 +    // Add public getter methods for each field if external access is required
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged